File: 1256-RankTransformOfAnArray/1256-RankTransformOfAnArray.py
# Last updated: 8/13/2025, 8:17:43 PM
import heapq
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        heap = []
        rank = 1

        for i in range(len(arr)) : 
            heapq.heappush(heap , (arr[i] , i))
        
        while len(heap) > 0 : 
            element , index = heapq.heappop(heap)
            arr[index] = rank
            if heap and heap[0][0] > element : 
                rank+=1
        
        return arr

File: 1256-RankTransformOfAnArray/1256-RankTransformOfAnArray.java
// Last updated: 8/13/2025, 8:17:50 PM
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int n = arr.length;
        HashMap<Integer,Integer> h = new HashMap();
        int[] sortedArray = Arrays.stream(arr).distinct().sorted().toArray();
        for(int i=0;i<sortedArray.length;i++){
            h.put(sortedArray[i],i+1);
        }

        for(int i=0;i<n;i++){
            arr[i] = h.get(arr[i]);
        }

        return arr;
    }
}